api_app/services/airlock.py

```diff
@@ -0,0 +1,97 @@
+    tre_id = config.TRE_ID
+    short_workspace_id = workspace.id[-4:]
+    if airlock_request.requestType == constants.IMPORT_TYPE:
+        if airlock_request.status == AirlockRequestStatus.Draft:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_EXTERNAL.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Submitted:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.InReview:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Approved:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_APPROVED.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.Rejected:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_IMPORT_REJECTED.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+    else:
+        if airlock_request.status == AirlockRequestStatus.Draft:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INTERNAL.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status in AirlockRequestStatus.Submitted:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INPROGRESS.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.InReview:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_INPROGRESS.format(short_workspace_id),
+                                         constants.WORKSPACE_RESOURCE_GROUP_NAME.format(tre_id, short_workspace_id))
+        elif airlock_request.status == AirlockRequestStatus.Approved:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_APPROVED.format(tre_id),
+                                         constants.CORE_RESOURCE_GROUP_NAME.format(tre_id))
+        elif airlock_request.status == AirlockRequestStatus.Rejected:
+            return RequestAccountDetails(constants.STORAGE_ACCOUNT_NAME_EXPORT_REJECTED.format(short_workspace_id),
```

Comment:
what happens if the current status doesnt match any of the if statements?

so for example we have 'blocking in progress' 'approval in progress', if you ask for the token when in this stage, what do you get?

I think, that if this is an 'unsupported' stage, just dont return anything